### ISSUE TITLE
Annotators lack an 'update' functionality

### ISSUE THREAD

**Issue by joerivandervelde** [molgenis/molgenis#4694]
There are some annotators you want to run multiple times to get the desired output. For instance, CADD scores may be retrieved from multiple source files or, for novel indels that need scoring, from a web service output file. However, annotators can only be ran once. A second run on the previous (lets say "half annotated" file) will result in all annotations of this type being dropped before being freshly annotated. This means you lose any values that do not have a match in the resource of a second annotation run.

For instance, consider the following file with "wrong" CADD scores:
```
22	25599848	.	C	T,TA	.	.	CADD=1,2;CADD_SCALED=3,4;
22	25599849	.	G	AT	.	.	.	CADD=5;CADD_SCALED=6;
22	25601188	.	A	G	.	.	CADD=7;CADD_SCALED=8;
```
When we run the CADD annotator, it simply re-annotates, dropping existing fields:
```
22	25599848	.	C	T,TA	.	.	CADD=6.956883,.;CADD_SCALED=33.0,.;
22	25599849	.	G	AT	.	.	.
22	25601188	.	A	G	.	.	CADD=3.177096;CADD_SCALED=22.7;
```
But with an 'update' mode, the annotator would leave them in place, allowing you to construct a more complete file:
```
22	25599848	.	C	T,TA	.	.	CADD=6.956883,2;CADD_SCALED=33.0,4;
22	25599849	.	G	AT	.	.	CADD=5;CADD_SCALED=6;
22	25601188	.	A	G	.	.	CADD=3.177096;CADD_SCALED=22.7;
```

**Comment by bartcharbon** [comment]
Interesting use case for this feature are the multiallelic variants.

if I have A,GGG and my first run yields "0.45,", and my second run returns ",0.55", you would like to end up with "0.45,0.55"

**Comment by fdlk** [comment]
Probably best solved together with version history of entity instances?

**Comment by fdlk** [comment]
Two questions:
* What is supposed to happen if an annotator updates an entity that already has a value specified?
* Does this mean the user can no longer trigger a "fresh" annotation if for whatever reason he doesn't like the annotations added with the previous one? (For instance, annotation done with buggy code, old version of a data file, a wrong genome build or whatever, yielding nonsense values)

**Comment by Mark-de-Haan** [comment]
Fixed in #4787
